app/services/decode_service.rb

```diff
@@ -0,0 +1,28 @@
+module DecodeService
+  def self.attach_image(file, target)
+    if Rails.env == 'test'
```

Comment:
not sure about this. In your request spec you are sending in a base64 encoded image

Reply:
When we are using this in production we are being sent a huge base64 string so we need to split it up. In the test environment here we are sending the image already split up into the needed keys. That is why we skip the `split_base64` method in the test environment.

Reply:
but again, you ARE sending in a base64 encoded image in a request spec.

Reply:
Yes, but I can skip the `split_base64` method because the object we pass in here in our test looks like this:
```
let(:image) do 
    {
      type: 'application/jpg',
      encoder: 'name=new_iphone.jpg;base64',
      data: 'iVBORw0KGgoAAAANSUhEUgAABjAAAAOmCAYAAABFYNwHAAAgAElEQVR4XuzdB3gU1cLG8Te9EEgISQi9I71KFbBXbFixN6zfvSiIjSuKInoVFOyIDcWuiKiIol4Q6SBVOtI7IYSWBkm',
      extension: 'jpg'
    }
```

Reply:
I understand now. Thx